src/compiler/parser.ts

```diff
@@ -1485,6 +1486,11 @@ module ts {
             identifierCount++;
             if (isIdentifier) {
                 let node = <Identifier>createNode(SyntaxKind.Identifier);
+
+                // Set strictModeKind property so that we can report appropriate error later in type checker
+                if (inStrictModeContext() && (token > SyntaxKind.Identifier && token <= SyntaxKind.LastFutureReservedWord)) {

```

Comment:
To make it clearer, I'd at least compare it to `FirstReservedWord`? Also, make your conditions as close to a 3-way inequality as possible (i.e. `blah < foo && foo <= bleep`).